ui/packages/app/src/project_setting/form/ProjectForm.js

```diff
@@ -57,41 +45,10 @@ const ProjectForm = () => {
     setName(newValue);
   };
 
-  const [streamError, setStreamError] = useState("");
-  const [isValidStream, setIsValidStream] = useState(false);
-  const onStreamChange = selectedStream => {
-    if (selectedStream !== project.stream) {
-      let isValid = isDNS1123Label(selectedStream);
-      if (!isValid) {
-        setStreamError(
-          "Stream name is invalid. It should contain only lowercase alphanumeric and dash (-)"
-        );
-      }
-      setIsValidStream(isValid);
-      setStream(selectedStream);
-    }
-  };
-
-  const [teamError, setTeamError] = useState("");
-  const [isValidTeam, setIsValidTeam] = useState(false);
-  const onTeamChange = selectedTeam => {
-    if (selectedTeam !== project.team) {
-      let isValid = isDNS1123Label(selectedTeam);
-      if (!isValid) {
-        setTeamError(
-          "Team name is invalid. It should contain only lowercase alphanumeric and dash (-)"
-        );
-      }
-      setIsValidTeam(isValid);
-      setTeam(selectedTeam);
-    }
-  };
-
-  useEffect(() => {
-    if (!project.team) {
-      setIsValidTeam(false);
-    }
-  }, [project.team]);
+  const [isValidStream, setIsValidStream] = useState(
+    isDNS1123Label(project.stream)
+  );
+  const [isValidTeam, setIsValidTeam] = useState(isDNS1123Label(project.team));
```

Comment:
You can simply set the states here as `false` by default, since all of the fields of `project` are uninitialised anyway (look at how these states were originally initialised before you started refactoring this file).